depsync/depsync.go

```diff
@@ -149,8 +174,57 @@ func isCursorExpired(err error) bool {
 	return strings.Contains(err.Error(), "EXPIRED_CURSOR")
 }
 
+func (w *watcher) processAutoAssign(devices []dep.Device) error {
```

Comment:
A lot of branching/relative complexity around these methods and state. Do you think this could be simplified further or maybe some of the logic could be extracted into functions?

Reply:
Hmm, I suppose I could break separate the actual assigner logic from the DEP message processing.